### PR TITLE
Add lavalink state (useful for resuming system)

### PR DESCRIPTION
**Pull request by flav28**

### CODE REVIEW

**Comment by Deivu** [comment]
Typings are not correctly specified

**Comment by KagChi** [comment]
Any benefit of this?

**Comment by flav28** [comment]
allows to know if the bot has been disconnected during a restart

**Comment by KagChi** [comment]
I dont get what you mean, this pr seems useless

**Comment by flav28** [comment]
if you don't have a system to summarize the players then it will be useless for you

**Comment by KagChi** [comment]
No i mean, this is not used internally by shoukaku it self. why dont you just extend the player instead?

File: src/guild/Player.ts [hunk 1]
```diff
@@ -576,6 +581,7 @@ export class Player extends EventEmitter {
     public onLavalinkMessage(json: any): void {
         if (json.op === OPCodes.PLAYER_UPDATE) {
             this.position = json.state.position;
+            this.lavalinkState = json.state.connected;
```
Comment (KagChi) [inline src/guild/Player.ts:584]: This is player connected state, missmatch with lavalinkState. should'nt playerState?

**Comment by flav28** [comment]
because it could be useful for other people and it doesn't impact the good functioning of shoukaku

**Comment by Deivu** [comment]
Instead of playerState, I'd call it connected, and do not make it type `boolean|nulll` instead make it only a boolean type that defaults to false until LL sends the first playerUpdate to this player. `playerState = true` is unclear on what it provides / says

**Comment by Deivu** [comment]
Could also be our own type, eg: type state = Connected | Disconnected | Unknown

**Comment by Deivu** [comment]
Closing due to inactivity, re-open if you applied the changes I said above

**Pull request closed without merging**